Add .deb support to ansible-opendaylight

AkshitaJha opened the pull request:
This commit checks the OS and then branches for either RPM or Deb
installs. This patch allows ODL to be installed in a debian distribution
(Debian or Ubuntu) via a deb repo.

Ansible version: 2.2.0.0
Tested for Debian 8.0 and Ubuntu 16.04

Signed-off-by: Akshita Jha <zenith158@gmail.com>

AkshitaJha committed:
Add .deb support to ansible-opendaylight

This commit checks the OS and then branches for either RPM or Deb
installs. This patch allows ODL to be installed in a debian distribution
(Debian or Ubuntu) via a deb repo.

Ansible version: 2.2.0.0
Tested for Debian 8.0 and Ubuntu 16.04

Signed-off-by: Akshita Jha <zenith158@gmail.com>

AkshitaJha committed:
Remove debug statements

Signed-off-by: Akshita Jha <zenith158@gmail.com>

dfarrell07 commented on vars/main.yml:
```diff
@@ -39,17 +39,24 @@ nb_rest_port: 8080
 #   rpm_path: Install ODL from a local path or remote URL
-install_method: "rpm_repo"
+#   dep_repo: Install ODL using a debian repository
+install_method: "deb_repo"
```
Can we leave the default the same? Else it's a backwards incompatible change.

AkshitaJha replied:
Done.

dfarrell07 commented on vars/main.yml:
```diff
@@ -39,17 +39,24 @@ nb_rest_port: 8080
 # NB: This will only take effect when `install_method` is "rpm_path"
 # Default to the Boron RPM hosted on the CentOS Community Build System
-rpm_path: "http://cbs.centos.org/repos/nfv7-opendaylight-5-release/x86_64/os/Packages/opendaylight-4.2.0-1.el7.noarch.rpm"
+rpm_path: "http://cbs.centos.org/repos/nfv7-opendaylight-5-release/x86_64/os/Packages/opendaylight-4.2.0-1.el7.noarch.rpm"
```
Nothing to do with your patch, but just saw that this line has a bug. Using 5-release repo to install ODL 4.2, which should fail.

dfarrell07 commented on tasks/add_odl_deb_repo.yml:
```diff
@@ -0,0 +1,18 @@
+- name: Add the repository key to apt
+  apt_key:
```
Cool, good to do this. Maybe should also be doing with RPM?

dfarrell07 commented on tasks/add_odl_deb_repo.yml:
```diff
@@ -0,0 +1,18 @@
+  apt_repository:
+    repo={{ deb_repo_url }}
+    state=present
+    filename='opendaylight'
```
Is this the normal file naming convention? The repo files end in .repo for yum, which is why I noticed.

AkshitaJha replied:
The source filenames end in .list. So, the repo is added in /etc/apt/sources.list.d/opendaylight.list.
    'filename' param docs: Sets the name of the source list file in sources.list.d. Defaults to a file name based on the repository source url. The .list extension will be automatically added.
See: http://docs.ansible.com/ansible/apt_repository_module.html

dfarrell07 replied:
Perfect, thanks

dfarrell07 commented on tasks/install_odl.yml:
```diff
@@ -1,5 +1,23 @@
 ---
+- set_fact:
+    os="{{ ansible_distribution }}_{{ ansible_distribution_major_version }}.0"
```
Some inline docs would be good here. For example, what ansible_distribution_version might typically resolve to. Also not clear without digging a bit why we need these facts. Also would be nice to doc the strangeness with x prefix to Ubuntu, and .0 suffix for Debian. Otherwise, I don't see a more clear/elegant way of differentiating those URLs.

AkshitaJha replied:
I have added some docs to help understand the variables better. Let me know if things are still not clear.

dfarrell07 reviewed (changes requested):
Awesome, looks great overall Akshita. Some minor stuff inline.

AkshitaJha committed:
Add .deb support to ansible-opendaylight

This commit checks the OS and then branches for either RPM or Deb installs.
It allows ODL to be installed in a debian based distribution (Debian or Ubuntu)
using a debian repo or .deb local/remote path.

Ansible version: 2.2.0.0
Tested for Debian 8.0 and Ubuntu 16.04

Signed-off-by: Akshita Jha <zenith158@gmail.com>

AkshitaJha committed:
Change `deb_path` to use remote URL instead of local path

Signed-off-by: Akshita Jha <zenith158@gmail.com>

dfarrell07 commented on tasks/install_odl.yml:
```diff
@@ -1,5 +1,47 @@
+# `ansible_distribution_version` param: The complete version of the OS
+# `ansible_distribution_major_version` param: The major version of the OS
+
+# The .debs are currently hosted in OpenSUSE Build Service (OBS).
```
Awesome docs, thanks

dfarrell07 reviewed (approved):
This is all wonderful Akshita, thank you so much for the quality work.
